### ISSUE TITLE
Undocumented Breaking Changes in V2.0

### ISSUE THREAD

**Issue by lockhartja** [sksamuel/aedile#33]
@sksamuel  First and foremost -- thanks for the effort you put into this library....really appreciate the effort,

Migrating to version 2 -- I can't seem to get the syntax of the eviction listener to pass the swiggle red line check (SRLC) -- attempt multiple variations; however, seems the combination of  `asCache` and eviction lamba definition do mix well....simply passing the readme example produces the following

<img width="752" alt="image" src="https://github.com/user-attachments/assets/abc3571e-9e61-4711-a214-e524b3b7bee8">

I only able to get reference to resolve explicitly specifying the generic <K, V> definition....and still breaks after the `asCache` extension function....any additional guidance to get this particular property to resolve correctly

**Comment by lockhartja** [comment]
After a little bit more digging -- guessing the signature now requires that we pass in a separate CoroutineScope for the eviction listener....am I under interpreting that correctly from this line?
https://github.com/sksamuel/aedile/blob/ef896798c1d81b2c8806120e7dd7b150c9dcae0e/aedile-core/src/main/kotlin/com/sksamuel/aedile/core/config.kt#L45

**Comment by sksamuel** [comment]
Thanks for flagging this up.
I missed updating those tests in 2.0.0.
Since the signatures clash from the builders in Caffeine and the extension functions, I've renamed them to withEvictionalListener and withRemovalListener, and those take the proper suspendable functions. They also no longer require a scope (you can provide one if desired).
I've released this as 2.0.1. Let me know if this all works for you.

**Comment by sksamuel** [comment]
Did you try withExpireAfter ? 

```
    Caffeine.newBuilder()
        .maximumSize(2000)
        .initialCapacity(500)
        .withExpireAfter(loggerExpiry)
        .asCache<Int,UDate>()
```

**Comment by lockhartja** [comment]
Apologies -- I missed that piece -- thanks for the prompt response....cheers!

**Comment by sksamuel** [comment]
Ok, glad it's all working for you !

**Comment by lockhartja** [comment]
@sksamuel Apologies for another ping, but is `withExpireAfter` a function in the current patched 2.0.1 version -- not seeing it in the code anywhere

**Comment by lockhartja** [comment]
> Can you paste your builder code.

I have tried these two variation of the code (see below), searching this repo I wasn't able to find a code definition for `withExpireAfter`, hence my initial question...thanks again for the look.

```
private val loggerExpiry =
  object : Expiry<Int, UDate> {
    override fun expireAfterCreate(key: Int, value: UDate, currentTime: Long) =
      (value.asInstance - Clock.System.now()).inWholeNanoseconds

    override fun expireAfterUpdate(
      key: Int,
      value: UDate,
      currentTime: Long,
      currentDuration: Long,
    ) = currentDuration

    override fun expireAfterRead(
      key: Int?,
      value: UDate?,
      currentTime: Long,
      currentDuration: Long,
    ) = currentDuration
  }

private val loggerCache =
  Caffeine.newBuilder()
    .maximumSize(2000)
    .initialCapacity(500)
    .expireAfter(loggerExpiry)
    .asCache<Int, UDate>() _<-- Does Not Resolve_

private val loggerCache =
  Caffeine.newBuilder()
    .maximumSize(2000)
    .initialCapacity(500)
    .withExpireAfter(loggerExpiry) _<-- Does Not Resolve_
    .asCache<Int, UDate>()

```

**Comment by sksamuel** [comment]
Ok it was some issue with the variance on the type parameters. Can you update to 2.0.2 and try again ?

**Comment by lockhartja** [comment]
@sksamuel Will do -- will look out for the release -- thanks again for the prompt attention

**Comment by lockhartja** [comment]
> It seems there is still no definition of `withExpireAfter`. `asCache` works as expected.
> 
> ```kotlin
> Caffeine.newBuilder()
>     .withExpireAfter(loggerExpiry)  <-- Does Not Resolve
> ```
> 
> Aedile version: 2.0.2

@endwaa the correct signature should be similar to:

```
private val loggerCache =
  Caffeine.newBuilder()
    .maximumSize(2000)
    .initialCapacity(500)
    .expireAfter(loggerExpiry)
    .asCache<Int, UDate>
```

**Comment by endwaa** [comment]
Yes, but no `asLoadingCache` available on `expireAfter`.

**Comment by sksamuel** [comment]
I must have been drinking to update asCache and forget about asLoadingCache (and bulk loading cache). I've released 2.0.3 with them all updated now.
